consensus/bor/heimdall/metrics.go

```diff
@@ -0,0 +1,82 @@
+package heimdall
+
+import (
+	"context"
+	"time"
+
+	"github.com/ledgerwatch/erigon/metrics"
+)
+
+type (
+	requestTypeKey struct{}
+	requestType    string
+
+	meter struct {
+		request map[bool]metrics.Meter // map[isSuccessful]metrics.Meter
+		timer   metrics.Timer
+	}
+)
+
+const (
+	stateSyncRequest       requestType = "state-sync"
+	spanRequest            requestType = "span"
+	checkpointRequest      requestType = "checkpoint"
+	checkpointCountRequest requestType = "checkpoint-count"
+)
+
+func withRequestType(ctx context.Context, reqType requestType) context.Context {
+	return context.WithValue(ctx, requestTypeKey{}, reqType)
+}
+
+func getRequestType(ctx context.Context) (requestType, bool) {
+	reqType, ok := ctx.Value(requestTypeKey{}).(requestType)
+	return reqType, ok
+}
+
+var (
```

Comment:
Let's comment these map of meters out as we'll add them once metrics are enabled in erigon later in a separate PR. Add a TODO to uncomment once metrics are added.